core/aws-core/src/main/java/software/amazon/awssdk/awscore/client/handler/AwsClientHandlerUtils.java

```diff
@@ -42,11 +42,9 @@ private AwsClientHandlerUtils() {
      * @return A bytebuffer representing the given request
      */
     public static ByteBuffer encodeEventStreamRequestToByteBuffer(SdkHttpFullRequest request) {
-        Map<String, HeaderValue> headers = request.headers()
-                                                  .entrySet()
-                                                  .stream()
-                                                  .collect(Collectors.toMap(Map.Entry::getKey, e -> HeaderValue.fromString(
-                                                      firstIfPresent(e.getValue()))));
+        Map<String, HeaderValue> headers = new LinkedHashMap<>();
```

Comment:
Anything preventing us to pre-size this hash map?

Reply:
Nope, good catch!